lib/pharos/ssh/remote_file.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'shellwords'
+
+module Pharos
+  module SSH
+    class RemoteFile
+      def initialize(client, path)
+        @client = client
+        @path = path.shellescape
+        freeze
+      end
+
+      def unlink
+        @client.exec!("rm #{@path}")
+      end
+
+      def write(content, sudo: true, append: false)
+        @client.exec!(
+          "#{'sudo ' if sudo}tee #{'--append ' if append}#{@path} > /dev/null",
+          stdin: content.respond_to?(:read) ? content : StringIO.new(content)
+        )
+      end
+
+      def append(content, sudo: true)
+        write(content, sudo: sudo, append: true)
+      end
+
+      def read
+        @client.exec!("sudo cat #{@path}")
+      end
+
+      def exist?
+        @client.exec?("[ -e #{@path} ]")
+      end
+
+      def with_existing
+        exist? && yield(self)
+      end
+
+      def download(local_path)
+        @session.download(@path, local_path)
+      end
+
+      def move(target)
+        @client.exec!("sudo mv #{@path} #{target.shellescape}")
+      end
+      alias mv move
+
+      def copy(target)
+        @client.exec!("sudo cp #{@path} #{target.shellescape}")
+      end
+      alias cp copy
+
+      def link(target)
+        @client.exec!("sudo ln -s #{@path} #{target.shellescape}")
```

Comment:
Not used, I guess it depends which way you think of it.

Something like:
```
file = ssh.file('/tmp/foo.sh')
file.write("#!/bin/sh\necho hello\n")
file.link('/usr/local/bin/hello')
```

Reply:
Yeah, I guess it can go either way, and it vaugely makes sense that `file.copy(foo)` and `file.link(foo)` both cause a new file to be created at `foo`.

Needs yardoc to clarify.